### PR TITLE
Network

### PR DESCRIPTION
**Pull request by hklerer**

### CODE REVIEW

**Comment by andrewoid** [comment]
Try adding enough so that you can actually run a unit test. Then we can talk.

File: src/main/java/klerer/rijks/ApiKey.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Retrieve the API Key in apikey.properties or from env variable on GitHub
+ */
+
+public class ApiKey {
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/ApiKey.java]: Please refer to the project email which has instructions for using this class, not copying the code into your project.

File: src/main/java/klerer/rijks/json/ArtObject.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package klerer.rijks.json;
+
+public class ArtObject {
+    public String title;
+    public String longTitle;
+    WebImage webImage;
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/json/ArtObject.java]: public

File: src/main/java/klerer/rijks/json/WebImage.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package klerer.rijks.json;
+
+public class WebImage {
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/json/WebImage.java:3]: Add the url

File: src/main/java/klerer/rijks/json/RijksCollection.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package klerer.rijks.json;
+
+public class RijksCollection {
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/json/RijksCollection.java:3]: You should use this.

File: src/main/java/klerer/rijks/RijksService.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+=import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+public interface RijksService {
+
+    @GET("/api/nl/collection")
+    Single<ArtObject> artObject(
+            @Query("key") String apikey,
+            @Query("p") int pageNum,
+            @Query("q") String num,
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksService.java]: q is not a num

File: src/main/java/klerer/rijks/RijksService.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+import io.reactivex.rxjava3.core.Single;
+import klerer.rijks.json.ArtObject;
+=import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+public interface RijksService {
+
+    @GET("/api/nl/collection")
+    Single<ArtObject> artObject(
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksService.java]: I asked for 3 methods, not one with all the fields.

File: src/test/java/klerer/rijks/RijksServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        // when
+        ArtObject artObject = service.artObject(
+                apiKey.toString(),
+                15,
+                "25",
+                "involved maker"
+        ).blockingGet();
+
+        // then
+        assertNotEquals(0, artObject.title);
```
Comment (andrewoid) [inline src/test/java/klerer/rijks/RijksServiceTest.java]: Title is a string and will never equal the int zero

File: src/main/java/klerer/rijks/ApiKey.java (hunk 1 above)
Comment (hklerer) [reply src/main/java/klerer/rijks/ApiKey.java]: am i not supposed to use this class? i read the email and could not find the instructions about using this class.

Comment (andrewoid) [reply src/main/java/klerer/rijks/ApiKey.java]: I never said to copy the class. 

Instructions are in the email

File: src/main/java/klerer/rijks/json/RijksCollection.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+
+    public ArtObject [] rijksCollection;
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/json/RijksCollection.java]: The names of variables in your classes must match the names of variables in the json

File: src/main/java/klerer/rijks/ApiKey.java (hunk 1 above)
Comment (hklerer) [reply src/main/java/klerer/rijks/ApiKey.java]: Am I supposed to use your ApiKey class from that link and make an EnviromentHandler, Handler, and PropertiesHandler?

Comment (andrewoid) [reply src/main/java/klerer/rijks/ApiKey.java]: Your supposed to add it as a dependency like it says in the instructions here
https://github.com/andrewoid/apikeys

File: src/test/java/klerer/rijks/RijksServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
+package klerer.rijks;
+
+import klerer.rijks.json.RijksCollection;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RijksServiceTest {
+    @Test
+    public void pageNum() {
```
Comment (andrewoid) [inline src/test/java/klerer/rijks/RijksServiceTest.java:11]: If these tests run locally, but don't run on GitHub then you haven't set up your API key on GitHub correctly.

Comment (hklerer) [reply src/test/java/klerer/rijks/RijksServiceTest.java:11]: my tests fail locally

File: .github/workflows/java_gradle.yaml [hunk 10]
```diff
@@ -46,7 +46,9 @@ jobs:
       - name: Grant execute permission for gradlew
         run: chmod +x gradlew
       - name: Build (and test) with Gradle
-        run: ./gradlew build -Dapikey=${{ env.APIKEY }}
+        env:
+          apikey: ${{ secrets.github_token }}
```
Comment (andrewoid) [inline .github/workflows/java_gradle.yaml]: This is not correct

File: src/main/java/klerer/rijks/RijksService.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+import io.reactivex.rxjava3.core.Single;
+import klerer.rijks.json.ArtObject;
+import klerer.rijks.json.RijksCollection;
+import retrofit2.http.GET;
+import retrofit2.http.Query;
+
+public interface RijksService {
+
+    @GET("/api/nl/collection")
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksService.java]: You want "en" not "nl"

File: src/main/java/klerer/rijks/RijksServiceFactory.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
+package klerer.rijks;
+
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RijksServiceFactory {
+    public RijksService getService() {
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://data.rijksmuseum.nl/")
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksServiceFactory.java]: Wrong baseUrl

File: src/test/java/klerer/rijks/RijksServiceTest.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    @Test
+    public void queryAndPageNum() {
+        // given
+        ApiKey apiKey = new ApiKey();
+        String keyString = apiKey.get();
+        RijksService service = new RijksServiceFactory().getService();
+
+        // when
+        RijksCollection rijksCollection = service.queryAndPageNum(keyString,
+                "query",
```
Comment (andrewoid) [inline src/test/java/klerer/rijks/RijksServiceTest.java]: This probably doesn't work. Put a real search in there

File: src/main/java/klerer/rijks/RijksServiceFactory.java [hunk 14]
```diff
@@ -0,0 +1,17 @@
+package klerer.rijks;
+
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RijksServiceFactory {
+    public RijksService getService() {
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://www.rijksmuseum.nl/en")
```
Comment (andrewoid) [inline src/main/java/klerer/rijks/RijksServiceFactory.java]: This isn't right. See the errors and look back at the url in their documentation.

**Review by andrewoid — APPROVED** [review]

**Comment by andrewoid** [comment]
I approved these changes, you should merge it.

**Pull request merged**